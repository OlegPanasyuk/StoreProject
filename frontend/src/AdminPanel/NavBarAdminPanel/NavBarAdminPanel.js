import React, { Component } from 'react';
import { Navbar, Nav, Container } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

export class NavBarAdminPanel extends Component {
    render() {
        let { match } = this.props;
        return (
            <Container>
                <Navbar>
                    <Nav>
                        <Nav.Item>
                            <NavLink className='p-3' exact to={`${match.url}/goods`}>Goods</NavLink>
                        </Nav.Item>
                        <Nav.Item>
                            <NavLink className='p-3' to={`${match.url}/users`}>Users</NavLink>
                        </Nav.Item>
                        <Nav.Item>
                            <NavLink className='p-3' to={`${match.url}/catalogue`}>Catalogue</NavLink>
                        </Nav.Item>
                    </Nav>
                </Navbar>
            </Container>
        );
    }
}

NavBarAdminPanel.propTypes = {
    match: PropTypes.object
};

export default NavBarAdminPanel;