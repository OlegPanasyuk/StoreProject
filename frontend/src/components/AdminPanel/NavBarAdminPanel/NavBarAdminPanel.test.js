import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import NavBarAdminPanel from './NavBarAdminPanel';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    match: {
        url: '/adminpanel'
    }
}) {
    const component = shallow(<NavBarAdminPanel {...props} />);
    return { component, props };
}

describe('NavBarAdminPanel --- Snapshot', () => {
    it('+++capturing Snapshot of ShoppingBasket', () => {
        const obj = setupComponent();
        expect(NavBarAdminPanel.propTypes.match).toBe(PropTypes.object);
        expect(shallowToJson(obj.component)).toMatchSnapshot();
    });
});
