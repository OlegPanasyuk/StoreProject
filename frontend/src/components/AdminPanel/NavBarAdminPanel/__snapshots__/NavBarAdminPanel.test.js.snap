// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarAdminPanel --- Snapshot +++capturing Snapshot of ShoppingBasket 1`] = `
<Bootstrap(undefined)
  style={
    Object {
      "backgroundColor": "var(--primary)",
    }
  }
>
  <Container
    as="div"
    fluid={false}
  >
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavItem
        as="div"
      >
        <NavLink
          className="p-3"
          exact={true}
          to="/adminpanel/goods"
        >
          Goods
        </NavLink>
      </NavItem>
      <NavItem
        as="div"
      >
        <NavLink
          className="p-3"
          to="/adminpanel/users"
        >
          Users
        </NavLink>
      </NavItem>
      <NavItem
        as="div"
      >
        <NavLink
          className="p-3"
          to="/adminpanel/catalogue"
        >
          Catalogue
        </NavLink>
      </NavItem>
      <NavItem
        as="div"
      >
        <NavLink
          className="p-3"
          to="/adminpanel/images"
        >
          Images
        </NavLink>
      </NavItem>
    </Nav>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Logout
    </Button>
  </Container>
</Bootstrap(undefined)>
`;
